### PR TITLE
[Compute] Restore point and Restore point collection

### PR DESCRIPTION
**Pull request by haagha**
## Description

Design Review
https://github.com/Azure/azure-powershell-cmdlet-review-pr/issues/938

## Checklist

- [x] I have read the [_Submitting Changes_](../blob/master/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](../blob/master/CONTRIBUTING.md)
- [x] The title of the PR is clear and informative
- [ ] The appropriate `ChangeLog.md` file(s) has been updated:
    - For any service, the `ChangeLog.md` file can be found at `src/{{SERVICE}}/{{SERVICE}}/ChangeLog.md`
    - A snippet outlining the change(s) made in the PR should be written under the `## Upcoming Release` header -- no new version header should be added
- [ ] The PR does not introduce [breaking changes](../blob/master/documentation/breaking-changes/breaking-changes-definition.md)
- [ ] If applicable, the changes made in the PR have proper test coverage
- [ ] For public API changes to cmdlets:
    - [ ] a cmdlet design review was approved for the changes in [this repository](https://github.com/Azure/azure-powershell-cmdlet-review-pr) (_Microsoft internal only_)
        - {Please put the link here}
    - [ ] the markdown help files have been regenerated using the commands listed [here](../blob/master/documentation/development-docs/help-generation.md#updating-all-markdown-files-in-a-module)

### CODE REVIEW

File: src/Compute/Compute/Generated/Models/PSRestorePoint.cs [hunk 1]
```diff
@@ -0,0 +1,33 @@
+﻿using Microsoft.Azure.Management.Compute.Models;
```
Comment (VeryEarly) [inline src/Compute/Compute/Generated/Models/PSRestorePoint.cs:1]: please add license to this file

File: src/Compute/Compute/Generated/Models/PSRestorePointCollection.cs [hunk 2]
```diff
@@ -0,0 +1,24 @@
+﻿using Microsoft.Azure.Management.Compute.Models;
```
Comment (VeryEarly) [inline src/Compute/Compute/Generated/Models/PSRestorePointCollection.cs:1]: please add license to this file

File: src/Compute/Compute/Generated/RestorePoints/GetAzRestorePoint.cs [hunk 3]
```diff
@@ -0,0 +1,60 @@
+﻿using System;
```
Comment (VeryEarly) [inline src/Compute/Compute/Generated/RestorePoints/GetAzRestorePoint.cs:1]: please add license to this file

File: src/Compute/Compute/Generated/RestorePoints/GetAzRestorePointCollection.cs [hunk 4]
```diff
@@ -0,0 +1,50 @@
+﻿using System;
```
Comment (VeryEarly) [inline src/Compute/Compute/Generated/RestorePoints/GetAzRestorePointCollection.cs:1]: please add license to this file

File: src/Compute/Compute/Generated/RestorePoints/NewAzRestorePoint.cs [hunk 5]
```diff
@@ -0,0 +1,83 @@
+﻿using System;
```
Comment (VeryEarly) [inline src/Compute/Compute/Generated/RestorePoints/NewAzRestorePoint.cs:1]: please add license to this file

File: src/Compute/Compute/Generated/RestorePoints/NewAzRestorePointCollection.cs [hunk 6]
```diff
@@ -0,0 +1,70 @@
+﻿using System;
```
Comment (VeryEarly) [inline src/Compute/Compute/Generated/RestorePoints/NewAzRestorePointCollection.cs:1]: please add license to this file

File: src/Compute/Compute/Generated/RestorePoints/RemoveAzRestorePoint.cs [hunk 7]
```diff
@@ -0,0 +1,70 @@
+﻿using System;
```
Comment (VeryEarly) [inline src/Compute/Compute/Generated/RestorePoints/RemoveAzRestorePoint.cs:1]: please add license to this file

File: src/Compute/Compute/Generated/RestorePoints/RemoveAzRestorePointCollection.cs [hunk 8]
```diff
@@ -0,0 +1,63 @@
+﻿using System;
```
Comment (VeryEarly) [inline src/Compute/Compute/Generated/RestorePoints/RemoveAzRestorePointCollection.cs:1]: please add license to this file

File: src/Compute/Compute/Generated/RestorePoints/UpdateAzRestorePointCollection.cs [hunk 9]
```diff
@@ -0,0 +1,64 @@
+﻿using System;
```
Comment (VeryEarly) [inline src/Compute/Compute/Generated/RestorePoints/UpdateAzRestorePointCollection.cs:1]: please add license to this file

**Review by VeryEarly — COMMENTED** [review]
Do you have design review accepted relate to this PR? Please provide if yes.
This new cmdlets are under generated folder but do not look like generated, please confirm.

**Comment by VeryEarly** [comment]
> Do you have design review accepted relate to this PR? Please provide if yes.
> This new cmdlets are under generated folder but do not look like generated, please confirm.

Please generate help markdown for new cmdlets as well

File: src/Compute/Compute/Generated/Models/PSRestorePoint.cs (hunk 1 above)
Comment (haagha) [reply src/Compute/Compute/Generated/Models/PSRestorePoint.cs]: added

File: src/Compute/Compute/Generated/Models/PSRestorePointCollection.cs (hunk 2 above)
Comment (haagha) [reply src/Compute/Compute/Generated/Models/PSRestorePointCollection.cs]: added

File: src/Compute/Compute/Generated/RestorePoints/GetAzRestorePoint.cs (hunk 3 above)
Comment (haagha) [reply src/Compute/Compute/Generated/RestorePoints/GetAzRestorePoint.cs]: added

File: src/Compute/Compute/Generated/RestorePoints/GetAzRestorePointCollection.cs (hunk 4 above)
Comment (haagha) [reply src/Compute/Compute/Generated/RestorePoints/GetAzRestorePointCollection.cs]: added

File: src/Compute/Compute/Generated/RestorePoints/NewAzRestorePoint.cs (hunk 5 above)
Comment (haagha) [reply src/Compute/Compute/Generated/RestorePoints/NewAzRestorePoint.cs]: added

File: src/Compute/Compute/Generated/RestorePoints/NewAzRestorePointCollection.cs (hunk 6 above)
Comment (haagha) [reply src/Compute/Compute/Generated/RestorePoints/NewAzRestorePointCollection.cs]: added

File: src/Compute/Compute/Generated/RestorePoints/RemoveAzRestorePoint.cs (hunk 7 above)
Comment (haagha) [reply src/Compute/Compute/Generated/RestorePoints/RemoveAzRestorePoint.cs]: added

File: src/Compute/Compute/Generated/RestorePoints/RemoveAzRestorePointCollection.cs (hunk 8 above)
Comment (haagha) [reply src/Compute/Compute/Generated/RestorePoints/RemoveAzRestorePointCollection.cs]: added

File: src/Compute/Compute/Generated/RestorePoints/UpdateAzRestorePointCollection.cs (hunk 9 above)
Comment (haagha) [reply src/Compute/Compute/Generated/RestorePoints/UpdateAzRestorePointCollection.cs]: added

**Comment by haagha** [comment]
Design Review 
Azure/azure-powershell-cmdlet-review-pr#938

Added license, and moved RestorePoints folder out of generated.

**Comment by haagha** [comment]
> Do you have design review accepted relate to this PR? Please provide if yes.
> This new cmdlets are under generated folder but do not look like generated, please confirm.

The cmdlets are manually written

File: src/Compute/Compute/RestorePoints/GetAzRestorePoint.cs [hunk 10]
```diff
@@ -0,0 +1,74 @@
...
+using Microsoft.Azure.Commands.Compute.Automation.Models;
+using Microsoft.Azure.Commands.Compute.Models;
+using Microsoft.Azure.Commands.ResourceManager.Common.ArgumentCompleters;
+using Microsoft.Azure.Management.Compute;
+using Microsoft.Azure.Management.Compute.Models;
+using Microsoft.WindowsAzure.Commands.Utilities.Common;
+
+namespace Microsoft.Azure.Commands.Compute.Automation
+{
+    [Cmdlet(VerbsCommon.Get, ResourceManager.Common.AzureRMConstants.AzureRMPrefix + "RestorePoint", DefaultParameterSetName = "DefaultParameter", SupportsShouldProcess = true)]
```
Comment (VeryEarly) [inline src/Compute/Compute/RestorePoints/GetAzRestorePoint.cs:29]:
```suggestion
    [Cmdlet(VerbsCommon.Get, ResourceManager.Common.AzureRMConstants.AzureRMPrefix + "RestorePoint", DefaultParameterSetName = "DefaultParameter")]
```

File: src/Compute/Compute/RestorePoints/GetAzRestorePointCollection.cs [hunk 11]
```diff
@@ -0,0 +1,64 @@
...
+using Microsoft.Azure.Commands.Compute.Automation.Models;
+using Microsoft.Azure.Commands.Compute.Models;
+using Microsoft.Azure.Commands.ResourceManager.Common.ArgumentCompleters;
+using Microsoft.Azure.Management.Compute;
+using Microsoft.Azure.Management.Compute.Models;
+using Microsoft.WindowsAzure.Commands.Utilities.Common;
+
+namespace Microsoft.Azure.Commands.Compute.Automation
+{
+    [Cmdlet(VerbsCommon.Get, ResourceManager.Common.AzureRMConstants.AzureRMPrefix + "RestorePointCollection", DefaultParameterSetName = "DefaultParameter", SupportsShouldProcess = true)]
```
Comment (VeryEarly) [inline src/Compute/Compute/RestorePoints/GetAzRestorePointCollection.cs:29]:
```suggestion
    [Cmdlet(VerbsCommon.Get, ResourceManager.Common.AzureRMConstants.AzureRMPrefix + "RestorePointCollection", DefaultParameterSetName = "DefaultParameter")]
```

**Review by VeryEarly — CHANGES_REQUESTED** [review]

**Comment by VeryEarly** [comment]
Hi @haagha , please these unnecessary attributes in comments, otherwise PR looks good.

File: src/Compute/Compute/RestorePoints/GetAzRestorePoint.cs (hunk 10 above)
Comment (haagha) [reply src/Compute/Compute/RestorePoints/GetAzRestorePoint.cs]: made the change

File: src/Compute/Compute/RestorePoints/GetAzRestorePointCollection.cs (hunk 11 above)
Comment (haagha) [reply src/Compute/Compute/RestorePoints/GetAzRestorePointCollection.cs]: made the change

**Comment by haagha** [comment]
@VeryEarly made the suggested changes, thanks.

**Comment by haagha** [comment]
@VeryEarly  Can you please advise on why the CI checks are failing, the issue doesn't seem to be from the code.

**Review by VeryEarly — APPROVED** [review]

**Pull request merged**